### ISSUE TITLE
Line break in C++ nested-name confuses doxygen

### ISSUE THREAD

**Issue by jwakely** [doxygen/doxygen#8633]
**Describe the bug**

Doxygen warns about this code:

```cpp
/// foo
template<typename T>
struct foo
{
  static constexpr T bar = 0;
};

/// definition of foo::bar
template<typename T>
constexpr T foo::
  bar;
```

**Expected behavior**
The line break between `foo::` and `bar` should have no effect (as is the case for the C++ grammar).

**To Reproduce**
Put the code above in `doc.cc` and create a `Doxyfile` that says `INPUT = doc.cc` then run `doxygen`

Doxygen will say:

```
...
Searching for member function documentation...
/tmp/doc.cc:10: warning: documented symbol 'constexpr T foo:: bar' was not declared or defined.
...
Generating docs for compound foo...
/tmp/doc.cc:5: warning: Member bar (variable) of struct foo is not documented.
```

**Version**

1.9.1

LSB Version:    :core-4.1-amd64:core-4.1-noarch
Distributor ID: Fedora
Description:    Fedora release 34 (Thirty Four)
Release:        34
Codename:       ThirtyFour

Using the default Fedora package, `doxygen-1.9.1-10.fc34.x86_64`

**Comment by jwakely** [comment]
This causes hundreds of lines of warnings for libstdc++ (and presumably, things missing from the docs).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8678

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
